File: Data Structures & Algorithms/duplicate-integer/submission-4.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        if ( nums.Length <= 1)
        {
            return false;
        }

        List<int> numSeen = new List<int>(); 

        for (int i = 0; i < nums.Length; i++) 
        {
            
            if (numSeen.IndexOf(nums[i]) == -1)
            {
                numSeen.Add(nums[i]);
            }
            else 
            {
                return true;
            }
        }

        return false; 

    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-4.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        // uses a rolling hashSet since we care about frequency / duplicate, within a range = sliding window also 

        HashSet<Integer> set = new HashSet<>(); 

        int l = 0; 
        for (int r = 0; r < nums.length; r++) {
            if (r - l > k) {
                set.remove(nums[l]);
                l++;
            }

            if (set.contains(nums[r])) {
                return true;
            }

            set.add(nums[r]);
        }

        return false;
    }
}